util/audio.py

```diff
@@ -1,31 +1,108 @@
 import os
+import io
 import sox
 import wave
+import opuslib
 import tempfile
 import collections
+import numpy as np
+
 from webrtcvad import Vad
+from util.helpers import LimitingPool
 
 DEFAULT_RATE = 16000
 DEFAULT_CHANNELS = 1
 DEFAULT_WIDTH = 2
 DEFAULT_FORMAT = (DEFAULT_RATE, DEFAULT_CHANNELS, DEFAULT_WIDTH)
 
+AUDIO_TYPE_NP = 'np'
+AUDIO_TYPE_PCM = 'pcm'
+AUDIO_FILE_PREFIX = 'audio/'
+AUDIO_TYPE_WAV = AUDIO_FILE_PREFIX + 'wav'
+AUDIO_TYPE_OPUS = AUDIO_FILE_PREFIX + 'opus'
+LOADABLE_FILE_FORMATS = [AUDIO_TYPE_WAV, AUDIO_TYPE_OPUS]
+
+OPUS_PCM_LEN_SIZE = 4
+OPUS_RATE_SIZE = 4
+OPUS_CHANNELS_SIZE = 1
+OPUS_WIDTH_SIZE = 1
+OPUS_CHUNK_LEN_SIZE = 2
+
+NP_TYPE_LOOKUP = [None, np.int8, np.int16, None, np.int32]
+UNSUPPORTED_TYPE = 'Unsupported audio type: {}'
+
+
+class Sample:
+    def __init__(self, audio_type, raw_data, audio_format=None):
```

Comment:
Maybe add some documentation? I don't understand the semantics of audio type and audio format, even after reading the code in this class.

Reply:
Some for the audio types are used for storing stuff (therefore I assigned them a "mime-type") while the others are just for internal use. How about renaming `LOADABLE_FILE_FORMATS` to `LOADABLE_AUDIO_TYPES` and remove the "audio/" part from the loadable ones?